docs/adr/0010-transition-fxa-from-backbone-to-react.md

```diff
@@ -0,0 +1,151 @@
+# Transition FxA from Backbone to React
+
+- Deciders: Lauren Zugai, Les Orchard, Danny Coates
+- Date: 2020-03-19
+
+## Context and Problem Statement
+
+When Firefox Accounts was created some 6-7 years ago, the front-end Javascript framework and HTML templating options were limited significantly to those available at the time. This ADR details the pros and cons of various FE libraries and frameworks, and why it is desirable for FxA to move to different technology. It does not cover _how_ we will begin the conversion which can be [read about here](https://github.com/mozilla/fxa/blob/master/docs/adr/0011-create-new-react-app-for-settings-redesign.md).
+
+While Backbone and Mustache have served their purposes for FxA thus far, other modern libraries and frameworks not only offer more appealing architecture and developer conveniences, but also offer better data predictability, potential performance boosts, and a strong community of online support.
```

Comment:
Good point. I think it's worth mentioning here that auth server uses mustache.

Might be good as a new issue with a `needs:discussion` label. It could be that it's fine TBH, but I've definitely found the _totally_ logic-less nature of mustache in the content server a little cumbersome. Probably depends on how annoying or limiting it is there. 😅

Reply:
> ... I wonder if there's another template system we should use someday ...

The popularity argument can be made in mustache's favor here.

Reply:
Mustache is probably fine for emails, but in case you're looking for a template language that is similar in syntax to Mustache, but slightly smarter (not by a lot, but far less annoying than Mustache) and more actively maintained, might I suggest [Liquid](https://github.com/docs/liquid).

Reply:
Of course if you wanted to use React wholesale, [this looks neat](https://github.com/chromakode/react-html-email).

Reply:
I am actually more bothered around the horrendous nature of our CSS in emails than templating language. We use `!important` everywhere.

I've never thought about Liquid outside of Shopify, lol. I was thinking EJS could also be a good one if we need logic, but since I have worked in emails sparingly, I'm not sure if it's worth the effort. Mustache seems to be sufficient for our needs.

I created a follow-up issue here where we can further the discussion: https://github.com/mozilla/fxa/issues/4627